server/src/server.rs

```diff
@@ -1146,12 +1146,17 @@ where
 				RpcServiceCfg::OnlyCalls,
 			));
 
-			Box::pin(
-				http::call_with_service(request, batch_config, max_request_size, rpc_service, max_response_size)
-					.map(Ok),
-			)
+			Box::pin(async move {
+				let rp =
+					http::call_with_service(request, batch_config, max_request_size, rpc_service, max_response_size)
+						.await;
+				drop(conn);
+				Ok(rp)
+			})
 		} else {
-			Box::pin(async { http::response::denied() }.map(Ok))
+			let rp = http::response::denied();
+			drop(conn);
+			Box::pin(async { Ok(rp) })
```

Comment:
I guess this is just a tidy up and isn't changing any semantics right?

Reply:
Yupp, it was to make it more verbose that it wasn't any async code but I can revert it if you want.

Reply:
I added a comment for this instead. I'm not super happy with the drop stuff here because it's easy to mess up but fine for now.